aloha-io-avro/src/test/scala/com/eharmony/aloha/audit/impl/avro/ImplicitsTest.scala

```diff
@@ -0,0 +1,121 @@
+package com.eharmony.aloha.audit.impl.avro
+
+import com.google.common.collect.Lists
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.BlockJUnit4ClassRunner
+
+import scala.collection.JavaConverters.seqAsJavaListConverter
+import com.eharmony.aloha.audit.impl.avro.Implicits.{RichFlatScore, RichScore}
+import java.{lang => jl, util => ju}
+
+import org.apache.avro.generic.GenericRecord
+
+/**
+  * Created by ryan.deak on 7/5/17.
+  */
+@RunWith(classOf[BlockJUnit4ClassRunner])
+class ImplicitsTest {
+  import ImplicitsTest._
+  import FlatScoreTest.flatScore
+
+  @Test def testFlatScoreToScore(): Unit =
+    assertEquals(score, flatScore.toScore)
+
+  @Test def testScoreToFlatScore(): Unit =
+    assertEquals(flatScore, score.toFlatScore)
+
+  /**
+    * Tests that scores are corrected inserted during conversions and that
+    * children ordering is preserved.
+    */
+  @Test def testRoundTripConversionFromScore(): Unit =
+    assertEquals(score, score.toFlatScore.toScore)
+
+  // TODO: Fix this test.  Once fixed, the stuff should be considered good.
+  @Test def testRoundTripConversionFromScoreIrregularTree(): Unit =
+    assertEquals(irregularTree, irregularTree.toFlatScore.toScore)
+
+  @Test def testRoundTripConversionFromFlatScore(): Unit =
+    assertEquals(flatScore, flatScore.toScore.toFlatScore)
+
+  /**
+    * Unlike `testRoundTripConversionFromScore`, this tests verifies that
+    * all fields in scores are correctly copied during conversions.
+    */
+  @Test def testAllFieldsAppear(): Unit = {
+    val s = filledInScore
+    assertEquals(s, s.toFlatScore.toScore)
+  }
+
+  @Test def testSameFieldsInGenericRecord(): Unit = {
```

Comment:
This test is important.  It shows that both `Score` and `FlatScore` have the same fields for the root element from the perspective of `GenericRecord`.